### PR TITLE
[Merged by Bors] - chore: change `InnerProductSpace.complexToReal` to a definition

### PR DESCRIPTION
**Pull request by xroblot**
The instance [docs#InnerProductSpace.complexToReal](https://leanprover-community.github.io/mathlib4_docs/find/?pattern=InnerProductSpace.complexToReal#doc) create some diamond with [docs#PiLp.innerProductSpace](https://leanprover-community.github.io/mathlib4_docs/find/?pattern=PiLp.innerProductSpace#doc), see [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Problem.20with.20.60EuclideanSpace.20.E2.84.82.20.CE.B9.60.20as.20.60InnerProductSpace.20.E2.84.9D.60/near/438183797). 

This PR changes this instance to a def instead and add the instance
```lean
instance : InnerProductSpace ℝ ℂ := innerProductSpace.complexToReal
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
!bench

File: Mathlib/Analysis/InnerProductSpace/Basic.lean [hunk 1]
```diff
@@ -2230,11 +2230,13 @@ set_option linter.uppercaseLean3 false in
 #align real_inner_I_smul_self real_inner_I_smul_self
 
 /-- A complex inner product implies a real inner product -/
```
Comment (sgouezel) [inline Mathlib/Analysis/InnerProductSpace/Basic.lean:2232]: Can you add a comment in the docstring explaining why it shouldn't be an instance?

**Comment by sgouezel** [comment]
A side remark: note that `!bench` is very expensive resourcewise (it triggers a full rebuild without using the cache), so you should only `!bench` things when you expect a serious performance change in one direction or the other one. Here, you are just disabling an instance, so it can't really have a performance impact -- what is possible is that things don't build. In any case, only do `!bench` when you know that things build, because otherwise it's even more wasteful.

File: Mathlib/Analysis/InnerProductSpace/Basic.lean (hunk 1 above)
Comment (xroblot) [reply Mathlib/Analysis/InnerProductSpace/Basic.lean]: I added a comment. Thanks also for the remark about bench!

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**